File: src/algorithms/unique-occurrences.js
/**
 * Given an array of integers arr, returns true if the number of occurrences
 * of each value in the array is unique, or false otherwise.
 * 
 * @param {number[]} arr - An array of numbers
 * @returns {boolean} - Returns true if the number of occurrences of each value
 *     in the array is unique
 */
function uniqueOccurrences (arr) {
  let numCount = {};
  let uniqueCount = new Set();
  for (const num of arr) {
    numCount[num] = (numCount[num] || 0) + 1;
  }
  for (const [key, val] of Object.entries(numCount)) {
    if (!uniqueCount.has(val)) {
      uniqueCount.add(val);
    } else {
      return false;
    }
  }
  return true;
}

File: src/algorithms/unique-occurrences.py
def unique_occurrences(arr):
    """Given an array of integers arr, returns true if the number of occurrences of each value in the array is unique or false otherwise.

    Args:
        arr (List[int]): A list of integers.

    Returns:
        Boolean: Returns true if the number of occurrences of each value in the array is unique or false otherwise.
    """
    elems = {}
    seen_vals = set()
    for num in arr:
        elems[num] = elems.get(num, 0) + 1
    for val in elems.values():
        if val in seen_vals:
            return False
        seen_vals.add(val)
    return True
